### ISSUE TITLE
Discrete parameters in emcee

### ISSUE THREAD

**Issue by Gabriel-p** [dfm/emcee#266]
Hi,

I'm having an issue sampling with `emcee` for a mix of discrete and continuous parameters. I currently follow a strategy very similar to that described in #150 by @lfloeer.

The thing is, I [opened a question](https://stats.stackexchange.com/questions/360076/discrete-and-continuos-parameters-in-mcmc-sampler) at *stats.stackexchange.com* about this yesterday, and I was advised that this approach was incorrect. I pointed the above issue to Eric (the *stats* user who responded this) and now he is also interested in why/how this works.

In my case, what I do is to "push" the sampled values for the discrete parameters to the closest "valid" value that parameter is allowed to have, before passing it to the likelihood (this is described in some more detail in the *stats* question linked above).

Is this a proper approach for handling discrete parameters?

**Comment by dfm** [comment]
Yeah - I think that the commenter on stackexchange misunderstood your suggestion. Take a look at my comment on #150 to see if it helps.

I definitely don't advocate for it as an efficient method, but it isn't fundamentally wrong 😄
